### PR TITLE
Implement the sse4.2 string comparison intrinsics

### PR DESCRIPTION
**Pull request by dlrobertson**
- Implement
   - _mm_cmpistrm
   - _mm_cmpistri
   - _mm_cmpistrz
   - _mm_cmpistrc
   - _mm_cmpistrs
   - _mm_cmpistro
   - _mm_cmpistra
   - _mm_cmpestrm
   - _mm_cmpestrz
   - _mm_cmpestrc
   - _mm_cmpestrs
   - _mm_cmpestro
   - _mm_cmpestra
 - Add documentation to _mm_cmpestri
 - Add missing constants
 - Fix typo in lib docs

I'm working on getting sse4.2 covered, but I figured this was a nice chunk to start with. Comments and critiques are welcomed!

### CODE REVIEW

**Comment by alexcrichton** [comment]
Looks great to me, thanks! I think @BurntSushi may want to take a look at the docs though to help out?

File: src/x86/sse42.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 use stdsimd_test::assert_instr;
 
 use x86::__m128i;
+use v128::*;
```
Comment (BurntSushi) [inline src/x86/sse42.rs]: Please arrange imports in alphabetical order. :-)

File: src/x86/sse42.rs [hunk 2]
```diff
@@ -15,11 +16,151 @@ pub const _SIDD_CMP_EQUAL_ORDERED: i8 = 0b00001100;
...
 pub const _SIDD_POSITIVE_POLARITY: i8 = 0b00000000;
 pub const _SIDD_NEGATIVE_POLARITY: i8 = 0b00010000;
+pub const _SIDD_MASKED_POSITIVE_POLARITY: i8 = 0b00100000;
 pub const _SIDD_MASKED_NEGATIVE_POLARITY: i8 = 0b00110000;
 
 pub const _SIDD_LEAST_SIGNIFICANT: i8 = 0b00000000;
 pub const _SIDD_MOST_SIGNIFICANT: i8 = 0b01000000;
 
+pub const _SIDD_BIT_MASK: i8 = 0b00000000;
+pub const _SIDD_UNIT_MASK: i8 = 0b01000000;
```
Comment (BurntSushi) [inline src/x86/sse42.rs]: We'll want to add docs to each of these constants, but I'm happy to punt on that for now!

Comment (dlrobertson) [reply src/x86/sse42.rs]: Added some short doc strings.

File: src/x86/sse42.rs (hunk 1 above)
Comment (dlrobertson) [reply src/x86/sse42.rs]: Good point.

**Comment by dlrobertson** [comment]
> I'm hoping we can improve the docs all around here since the packed string compare functions have a lot of interesting semantics and modes, and docs for them are rather sparse.

100% agree. I'd like to circle back at some point and add some examples etc.

> One thing I think would be nice here would be to improve the tests by using actual strings. I think the first test started out that way, just because it seems more intelligible to me... Would you want to try to do that in this PR?

I started by using strings, but I found it easier to debug when manually specifying the ASCII or UTF-16 bytes. Should be a simple change to use strings instead of the raw bytes. I can definitely include that in this PR.

**Comment by BurntSushi** [comment]
@dlrobertson Awesome, thanks! (Maybe I'm wrong that strings are easier?)

**Comment by BurntSushi** [comment]
@dlrobertson Nice work on the tests! They look much better. :-) I'll merge once CI passes.

**Pull request merged**